### PR TITLE
Avoid creating new media item when unnecessary

### PR DESCRIPTION
**Pull request by elibud**
**Fixes** #7344 

**To test:**
1. Select one site
2. Tap on its icon
3. Select Change Icon
4. Select the Media Library
5. Chose a square item and do not resize it nor crop it
6. Once the item is set as the site icon, check on the Media Library that there is not a duplicate instance of it.
7. Do the same steps but modifying the media item, check that a new media item is created.
8. Do the same steps but choosing an image from your camera roll, check that a new media item is created.

Needs review: @frosty

### CODE REVIEW

**Comment by elibud** [comment]
@frosty ready for another look

File: WordPress/Classes/ViewRelated/Blog/BlogDetailsViewController.m [hunk 1]
```diff
@@ -643,13 +643,21 @@ - (void)updateSiteIcon
...
-        if (image) {
-            [weakSelf siteIconImageSelected:image];
+    self.siteIconPickerPresenter.onCompletion = ^(Media *media, NSError *error) {
+        if (media == nil && error == nil) {
+            // If no media and no error the picker was cancelled
+            [weakSelf dismissViewControllerAnimated:YES completion:nil];
+        } else if (media && error == nil) {
+            [weakSelf updateBlogIconWithMedia:media];
+        } else if (error != nil) {
+            [weakSelf showErrorForSiteIconUpdate];
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Blog/BlogDetailsViewController.m]: I think you can simplify this a little?

```
if (error) {
    [weakSelf showErrorForSiteIconUpdate];
} else if (media) {
    [weakSelf updateBlogIconWithMedia:media];
} else {
    [dismissViewControllerAnimated:YES completion:nil];
}
```

File: WordPress/Classes/ViewRelated/Blog/SiteIconPickerPresenter.swift [hunk 2]
```diff
@@ -11,7 +11,11 @@ class SiteIconPickerPresenter: NSObject {
     /// MARK: - Public Properties
 
     var blog: Blog
-    var onCompletion: ((UIImage?) -> Void)?
+    /// Will be invoked with a newly created image OR an existing media item to set
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Blog/SiteIconPickerPresenter.swift]: Need to update this comment.

File: WordPress/Classes/ViewRelated/Blog/BlogDetailsViewController.m (hunk 1 above)
Comment (elibud) [inline WordPress/Classes/ViewRelated/Blog/BlogDetailsViewController.m]: I like :)

**Pull request merged**